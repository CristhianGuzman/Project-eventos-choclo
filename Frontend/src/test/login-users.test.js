import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { createSerializer } from 'enzyme-to-json'

Enzyme.configure({ adapter: new Adapter()})
expect.addSnapshotSerializer( createSerializer({ mode: 'deep'}))

import LoginUser from '../components/Login/login-user'
import { shallow } from 'enzyme'

describe('Pruebas sobre el componente Login usuario', ()=>{

    //Snapshot
    test('Probando con enzyme', ()=>{
        const wrapper = shallow(<LoginUser/>)
        expect(wrapper).toMatchSnapshot()
    })

    test('Verificar mensajes de error en inputs', ()=>{
        const wrapper = shallow(<LoginUser/>)
        console.log(wrapper.debug())
        expect(wrapper.find('.invalid-feedback').length).toBe(2);
      
    })

    /*test('Verificar boton 1', ()=>{
        const wrapper = shallow (<LoginUser/>)
        const boton1 = wrapper.find('#boton1').at(0).simulate('click')
        expect(boton1).toBeDefined();

    })*/

    
})
