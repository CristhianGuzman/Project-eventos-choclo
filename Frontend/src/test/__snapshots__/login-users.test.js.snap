// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente Login usuario Probando con enzyme 1`] = `
<Fragment>
  <Menu />
  <hr />
  <div
    className="container-fluid "
  >
    <div
      className="modal-dialog text-center"
    >
      <div
        className="col-sm-9 main-section"
      >
        <div
          className="modal-content bg-warning"
        >
          <div
            className="col-12 user-img"
          >
            <i
              className="fas fa-users fa-6x d-flex justify-content-center"
              style={
                Object {
                  "color": "#ffffff",
                }
              }
            />
            <h3
              className="mb-3"
            >
              Bienvenido,
            </h3>
            <h6
              className="mb-3 align-items-start"
            >
              Por favor autentíquese para ingresar
            </h6>
            <form
              onSubmit={[Function]}
            >
              <div
                className="form-group"
                id="#user-group"
              >
                <label>
                  Usuario
                </label>
                <input
                  className="form-control  "
                  id="usuario"
                  name="usuario"
                  onChange={[Function]}
                  placeholder="Usuario"
                  type="text"
                />
                <div
                  className=" mb-3 invalid-feedback"
                >
                  Ingrese el usuario
                </div>
              </div>
              <div
                className="form-group"
                id="contrasena-group"
              >
                <label>
                  Contraseña
                </label>
                <input
                  className="form-control  "
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Contraseña"
                  type="password"
                />
                <div
                  className="mb-3 invalid-feedback"
                >
                  Ingrese la contraseña
                </div>
              </div>
              <h6
                className=" mb-3"
              >
                ¿Olvidó su contraseña?
              </h6>
              <div
                className="row justify-content-center align-items-center "
              >
                <button
                  className="btn btn-success col-3 mx-3"
                  id="boton1"
                  type="submit"
                >
                  Ingresar
                </button>
                <button
                  className="btn btn-success col-4"
                  id="boton2"
                  onClick={[Function]}
                  type="button"
                >
                  Registrarse
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
  <Footer />
</Fragment>
`;
